tdms/CMakeLists.txt

```diff
@@ -81,6 +81,17 @@ else()
     add_definitions(-DSPDLOG_ACTIVE_LEVEL=SPDLOG_LEVEL_INFO)
 endif()
 
+
+# TDMS version ----------------------------------------------------------------
+# if supplied via the cmake configuration (-DTDMS_VERSION=whatever_I_want) then
+# respect that, otherwise construct a version from a git tag or commit hash.
+if (NOT TDMS_VERSION)
```

Comment:
One slightly annoying thing is that `-DTDMS_version` will be persistent between builds, and always take priority. 

This isn't a super big problem (since I presume we're always using fresh builds to test new features) but it does mean that if you provide a manual version number _once_ then you have to clear that variable from your `CMakeCache` manually. Is there a way to tell `cmake` to clear the manually stored variable / not place it in the cache?

Reply:
Not without breaking the build. It needs to be in the cache (otherwise it won't get propagated at compile).

By-the-way this "problem" will also affect the release names in general... because it always needs to be cached. So you can break my system if you make a new commit and compile directly without rerunning the CMake configuration step, the "release" will be `<branchname>_<old commit hash>` for exactly the same reason.

Reply:
No problem, we'll just have to remember to do clean builds when developing if we care about the version name. End-users should never encounter the problem if they're building from main and following our setup instructions anyway.